nh_clinical/activity_extension.py

```diff
@@ -538,110 +530,241 @@ def get_activity_user_ids(self, cr, uid, activity_id, context=None):
         user_ids += follower_ids
         return list(set(user_ids))
 
-    # TODO EOBS-703: Trigger policy method is too large
-    def trigger_policy(self, cr, uid, activity_id, location_id=None,
-                       case=False, context=None):
+    # TODO EOBS-2171 This can be removed?
+    @api.model
+    def check_policy_activity_context(self, activity, location_id=None):
+        """
+        Check that the policy activity can be triggered based on the location's
+        context (Normally eobs)
+
+        :param activity: Policy activity definition
+        :type activity: dict
+        :param location_id: ID of location to check policy for
+        :type location_id: int
+        :return: True if can create activity for location
+        :rtype: bool
+        """
+        location_model = self.env['nh.clinical.location']
+        triggered_context = activity.get('context')
+        if triggered_context and location_id:
+            location = location_model.browse(location_id)
+            # Create a list of bools representing if the activity and location
+            # contexts match. If none of them match then return False
+            if not any(
+                    [c.name == triggered_context for c in
+                     location.context_ids]):
+                return False
+        return True
```

Comment:
I was confused what this method was doing for ages because of misleading variable names. At first I didn't realise that `context` is something in the policy dictionary, nothing to do with the Odoo context. Then I thought that activity was an `nh.activity` record and took a while to clock that its a part of the policy dictionary. Is there some other terminology we could use?

Reply:
You've used the term `activity_dict` in other methods I think.

Reply:
changed